### ISSUE TITLE
Realtime motiongram preview in the UI? 

### ISSUE THREAD

**Issue by aleksati** [balintlaczko/VideoAnalysis#50]
Many videos need radically different noise reduction parameters (especially threshold-values) to make decent looking motiongrams. And now, choosing the type of noise reduction and to print in color or not will affect the aesthetics of the exported image to a huge degree. In this case, I think it would be valuable for a user to be able to preview a motiongram in the UI when doing playback in realtime.

I think this makes a lot of sense in terms of user-friendliness. Might be more CPU-demanding though. Any thoughts on this?

We could also combine the "history image" and "motiongram" box in the UI to one box called "motion images".

**Comment by balintlaczko** [comment]
Could be nice, but I see a challenge of implementation when the user does not only using straight playback, but hopping over different time points with the playbar to check how things work. Could be nice with a "short-term" motiongram which rolls like a multislider (and then the discontinuity due to hopping is less of an issue). 
I don't think the CPU is a big problem as long as we keep it optional and cut off the processing when not in demand.
As for the other change, I like the idea to be able to control what exactly I am exporting, but I guess this app is for beginner users, so having less options is probably helpful not to get lost.

**Comment by aleksati** [comment]
Yes, I see. But I think this would be very nice actually. I´m going to make a demo of it and then we can look at it together and just discard it if it turns out to be messy, redundant or just not optimal for the app.

**Comment by aleksati** [comment]
This realtime mg-preview function is now implemented. It´s based on the time and duration of the video rather then framedump values. Made it so that scrolling the playbar clears the frames that it scrolls over, together with a toggle to clear the whole window. However, the scrolling function doesn't really work that well in the patch, due to lagging of the realtime playback. In the separate patch where I built it, where there was no realtime playback lag, the function worked pretty well. It might be more equipt for realtime video-apps.. Anyway, we´ll see if we incorporate the function in the final patch in January.
